### DISCUSSION TITLE
Vintage console popups + McGugan boxes?

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#14752] [Show and tell]
@willmcgugan did something pretty cool and [figured out a new way to draw boxes in TUIs](https://www.willmcgugan.com/blog/tech/post/ceo-just-wants-to-draw-boxes/). What if conhost used that for drawing popups?

### Before

![image](https://user-images.githubusercontent.com/189190/215223340-12b6589a-1a5b-444d-a7a3-c14aba3e86d0.png)
![image](https://user-images.githubusercontent.com/189190/215223357-63464fe9-a301-4c89-a944-f0f2bb178cd8.png)

### After

![image](https://user-images.githubusercontent.com/189190/215223299-9eec6fcf-91b6-4410-bbe1-8a28b8f94dea.png)
![image](https://user-images.githubusercontent.com/189190/215223426-c0d7edde-c288-49b0-81da-eec21d0d536f.png)

**Comment by zadjii-msft** [comment]
* People would rake us over the coals that we trumped their user settings in pursuit of being more visually appealing
* Some oil rig would stop working
* {your least favorite OEM} would complain that their tests that literally test if the buffer has line-drawing chars in it would explode and file a p0 drop everything "we need to fix their product line" bug

lets go for it

**Comment by j4james** [comment]
The problem is that McGugan boxes don't work very well when popped up over other text, which is kind of the main purpose of these windows. Comparing these two views, I'd definitely prefer the one on the right.

![image](https://user-images.githubusercontent.com/4181424/215225326-c7830b1f-0174-4eea-9b6a-371d00faed37.png)

I suppose it might look better with an extra blank border on the left and right, but it feels then like it's just wasting space.

**Reply by zadjii-msft** [reply]
That's a _very_ good point. 

Could we fake it with overlines/underlines? They probably wouldn't line up, would they...

**Reply by DHowett** [reply]
I actually tried the underline/overline/left and right lines a couple years ago! They're not feasible for ConPTY transit at the moment, and the use of reverse text makes it hard to ensure consistent styling.

You're right, of course - this is not a shippable solution... but a fun one to think about!

**Reply by j4james** [reply]
Conhost could probably do it with gridlines, because the bottom gridline is truly on the bottom, but a regular underline wouldn't work. And if you're going to use gridlines, you might as well use them for the whole border, and not bother with the fancy Unicode chars.

**Reply by j4james** [reply]
Another option, if you don't care too much about cross-terminal compatibility, would be to use soft fonts. That way you can design a border exactly how you want it. For example, something like this:

![image](https://user-images.githubusercontent.com/4181424/215229068-e3a28a1e-6293-488e-a064-be49d1738df3.png)

That's not really practical for the console popups, though.
